src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -154,6 +156,149 @@ public static void MergeAttributes(this TagHelperOutput tagHelperOutput, TagBuil
             }
         }
 
+        /// <summary>
+        /// Adds the given <paramref name="classValue"/> to the <paramref name="tagHelperOutput"/>'s
+        /// <see cref="TagHelperOutput.Attributes"/>.
+        /// </summary>
+        /// <param name="tagHelperOutput">The <see cref="TagHelperOutput"/> this method extends.</param>
+        /// <param name="classValue">The class value to add.</param>
+        /// <param name="htmlEncoder">The current HTML encoder.</param>
+        public static void AddClass(
+            this TagHelperOutput tagHelperOutput,
+            string classValue,
+            HtmlEncoder htmlEncoder)
+        {
+            if (tagHelperOutput == null)
+            {
+                throw new ArgumentNullException(nameof(tagHelperOutput));
+            }
+
+            var encodedClassValue = htmlEncoder.Encode(classValue);
+
+            if (SpaceChars.Any(encodedClassValue.Contains))
```

Comment:
The `HtmlEncoder` is likely to encode the HTML spaces other than `'\u0020'` because they're in the Unicode Cc category (Other, Control). This would be safer above the `Encode(...)` call.

Reply:
Handle `!tagHelperOutput.Attributes.TryGetAttribute("class", ...)` and `string.IsNullOrEmpty(currentClassValue)` before encoding. No need for any up-front encoding in either case i.e. avoid using `classValue` again once `encodedClassValue` exists.